packages/pluggableWidgets/timeline-web/src/components/TimelineComponent.tsx

```diff
@@ -0,0 +1,88 @@
+import { RenderModeEnum } from "../../typings/TimelineProps";
+import { Children, createElement, ReactElement, ReactNode } from "react";
+import classNames from "classnames";
+import { BasicItemType, CustomItemType, ItemType } from "../Timeline";
+import { Icon } from "mendix/components/web/Icon";
+
+interface TimelineProps {
+    data: Map<string | ReactNode, ItemType[]>;
+    renderMode: RenderModeEnum;
+    showDayDivider: boolean;
+}
+
+export default function TimelineComponent(props: TimelineProps): ReactElement {
+    return <div className="timeline-wrapper">{getItems(props.data, props.renderMode, props.showDayDivider)}</div>;
+}
+
+function getBasicEventsFromDay(eventsOfDay: BasicItemType[]): ReactNode[] {
+    return eventsOfDay.map((event, index) => (
+        <li className="timeline-event" key={index}>
+            <div className={"icon-wrapper"}>
+                {event.icon?.value ? <Icon icon={event.icon.value} /> : <div className="timeline-icon-circle" />}
+            </div>
+            <div className="flex-container content-wrapper">
+                <div className="date-time-wrapper">
+                    <p>{event.time}</p>
+                </div>
+                <div className="flex-container info-wrapper">
+                    <p className="title">{event.title}</p>
+                    <p className="description">{event.description}</p>
+                </div>
+            </div>
+        </li>
+    ));
+}
+
+function getCustomEventsFromDay(eventsOfDay: CustomItemType[]): { eventDayDivider: ReactNode; events: ReactNode[] } {
+    let eventDayDivider: ReactNode = null;
+    const events = eventsOfDay.map((event, index) => {
+        eventDayDivider = event.dayDivider;
+        return (
+            <li className="timeline-event" key={index}>
+                <div className="icon-wrapper">{event.icon ?? <div className="timeline-icon-circle" />}</div>
+                <div className="flexcontainer content-wrapper">
+                    {Children.count((event.eventDateTime as ReactElement).props.children) > 0 && (
```

Comment:
If we use renderer method we can always use ReactElement and prevent this casting